cpp/src/slice2cs/Gen.cpp

```diff
@@ -1918,122 +1897,105 @@ Slice::Gen::TypesVisitor::visitExceptionEnd(const ExceptionPtr& p)
     DataMemberList classMembers = p->classDataMembers();
     DataMemberList optionalMembers = p->orderedOptionalDataMembers();
     ExceptionPtr base = p->base();
-    string exParam = getEscapedParamName(allDataMembers, "innerException");
-
-    // Primary constructor.
-    _out << sp;
-    emitGeneratedCodeAttribute();
-    _out << nl << "public " << name << spar;
 
-    vector<string> paramDecl;
-    vector<string> secondaryCtorParams;
-    vector<string> secondaryCtorMemberNames;
-    vector<string> secondaryCtorBaseParamNames;
-    bool hideParameterlessCtor = false;
-
-    for (const auto& q : allDataMembers)
+    if (!allDataMembers.empty())
     {
-        string memberName = fixId(q->name(), DotNet::Exception);
-        string memberType = typeToString(q->type(), ns, q->optional());
-        paramDecl.push_back(memberType + " " + memberName);
+        // Primary constructor.
```

Comment:
This is largely a copy&paste from the class code.

Unfortunately, we can't move this code to a common function between ClassDefPtr and ExceptionPtr are unrelated types.